sound/soc/sof/ipc3.c

```diff
@@ -39,7 +40,7 @@ static const struct ipc_pm_ops ipc3_pm_ops = {
 };
 
 const struct ipc_ops ipc3_ops = {
-	.tplg = &ipc3_tplg_ops,
+	.tplg = &ipc4_tplg_ops,
```

Comment:
I really don't get this, even in a HACK :-)

Reply:
> I really don't get this, even in a HACK :-)

We test
```
	if (!ipc->ops->pcm || !ipc->ops->tplg || !ipc->ops->tplg->widget ||
	    !ipc->ops->tplg->control) {
```
in [snd_sof_ipc_init](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/ipc.c#L1014)
Anyway, I will remove the ` !ipc->ops->tplg->control` locally.